examples/coreset_in_model_pipeline.py

```diff
@@ -0,0 +1,243 @@
+    x, x_test, y, y_test = train_test_split(x, y, test_size=0.1, random_state=seed)
+
+    # Scale the features and responses individually
+    feature_scaler = StandardScaler().fit(x)
+    x = jnp.asarray(feature_scaler.transform(x))
+    x_test = jnp.asarray(feature_scaler.transform(x_test))
+
+    response_scaler = StandardScaler().fit(y)
+    y = jnp.asarray(response_scaler.transform(y))
+    y_test = jnp.asarray(response_scaler.transform(y_test))
+
+    # Request 200 coreset points
+    coreset_size = 200
+
+    # Setup the original data object
+    data = Data(x)
+
+    # Set the bandwidth parameter of the kernel using a median heuristic derived from at
+    # most 1000 random samples in the data.
+    num_samples_length_scale = min(num_data_pairs, 1_000)
+    generator = np.random.default_rng(seed)
+    idx = generator.choice(num_data_pairs, num_samples_length_scale, replace=False)
+    length_scale = median_heuristic(x[idx]).item()
+    kernel = SquaredExponentialKernel(length_scale=length_scale)
+
+    print("Estimating kernel ridge regression model with full data...")
+    full_data_fit_time = time()
+    full_data_model = KernelRidge(kernel="rbf", gamma=length_scale, alpha=1e-2).fit(
+        x, y
+    )
+    full_data_fit_time = time() - full_data_fit_time
+    full_data_mse = jnp.linalg.norm(y_test - full_data_model.predict(x_test))
+
+    print("Computing herding coreset...")
+    # Compute a coreset using kernel herding with a squared exponential kernel.
+    coreset_solver = KernelHerding(coreset_size, kernel)
+
+    coreset_build_time = time()
+    coreset, _ = eqx.filter_jit(coreset_solver.reduce)(data)
+    coreset_build_time = time() - coreset_build_time
+
+    print("Estimating kernel ridge regression model with coreset...")
+    coreset_indices = coreset.unweighted_indices
+    coreset_fit_time = time()
+    coreset_model = KernelRidge(
+        kernel="rbf", gamma=float(median_heuristic(x[coreset_indices])), alpha=1e-1
+    ).fit(x[coreset_indices], y[coreset_indices])
+    coreset_fit_time = time() - coreset_fit_time
+    coreset_overall_time = coreset_build_time + coreset_fit_time
+    coreset_mse = jnp.linalg.norm(y_test - coreset_model.predict(x_test))
+
+    print("Estimating kernel ridge regression model with random sample of data...")
+    solver = RandomSample(coreset_size, jr.key(seed))
+    random_set, _ = eqx.filter_jit(solver.reduce)(data)
+    random_indices = random_set.unweighted_indices
+    random_fit_time = time()
+    random_model = KernelRidge(
+        kernel="rbf", gamma=float(median_heuristic(x[random_indices])), alpha=1e-1
+    ).fit(x[random_indices], y[random_indices])
+    random_fit_time = time() - random_fit_time
+    random_mse = jnp.linalg.norm(y_test - random_model.predict(x_test))
+
+    # Produce some scatter plots
+    x_plot = jnp.linspace(x.min(), x.max(), 100).reshape(-1, 1)
+    plt.scatter(x, y, s=20, color="black", alpha=0.25, label="Data")
```

Comment:
I think it's worth adding a comment before each plot and a line break between them, so there's a bit more visual separation between the code for the different plots.